test/Microsoft.AspNet.Mvc.DataAnnotations.Test/Internal/DataAnnotationsModelValidatorProviderTest.cs

```diff
@@ -99,13 +100,24 @@ public void IValidatableObjectGetsAValidator()
             var mockValidatable = new Mock<IValidatableObject>();
             var metadata = _metadataProvider.GetMetadataForType(mockValidatable.Object.GetType());
 
-            var providerContext = new ModelValidatorProviderContext(metadata);
+            var providerContext = new ModelValidatorProviderContext(metadata, GetValidatorItems(metadata));
```

Comment:
Spoke offline. We are not doing this because `context.Results` will contain items added by a previous provider and creating a new context with a fresh set of items is not very common.